pipeline/sign/sign.go

```diff
@@ -0,0 +1,186 @@
+// Package sign provides a Pipe that signs .checksums files.
+package sign
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strconv"
+	"time"
+
+	"github.com/apex/log"
+	"github.com/mitchellh/go-homedir"
+	"golang.org/x/crypto/openpgp"
+	"golang.org/x/crypto/ssh/terminal"
+	"golang.org/x/sync/errgroup"
+
+	"github.com/goreleaser/goreleaser/context"
+	"github.com/goreleaser/goreleaser/pipeline"
+)
+
+type Pipe struct{}
+
+func (Pipe) String() string {
+	return "Signing release artifacts"
+}
+
+func (Pipe) Run(ctx *context.Context) (err error) {
+	if ctx.Config.Sign.GPGKeyID == "" {
+		return pipeline.Skip("sign.gpg_key_id is not configured")
+	}
+	return gpgSign(ctx)
+}
+
+func gpgSign(ctx *context.Context) error {
+	rawKeyID := ctx.Config.Sign.GPGKeyID
+
+	if len(rawKeyID) != 16 {
+		return fmt.Errorf("invalid gpg_key_id %q, needs to be a 8 byte long hex key id", rawKeyID)
+	}
+
+	keyID, err := strconv.ParseUint(rawKeyID, 16, 64)
+	if err != nil {
+		return fmt.Errorf("invalid gpg_key_id '%s': %s", rawKeyID, err)
+	}
+
+	// todo(fs): is this always in that location?
```

Comment:
no... it can be Defined using the `secret-keyring <dir>/secring.gpg` config...

https://www.gnupg.org/gph/en/manual/r1593.html

Reply:
but I think we can deal with this when someone needs it :)